HPCC-17846 DOCS:Additional Info for Multiple Thors

g-pan opened the pull request:
Fix HPCC-17846 DOCS:Additional Info for Multiple Thors
Adding additional information to the section about running multiple Thors
on the same cluster in the System Administrators book.

@jakesmith please review this content as it was taken from an email thread of a user's question.
@JamesDeFabia  please review

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

Test results (and PDF) can be viewed at http://10.240.32.243/view/Docs/job/DocDev0V/17/CE=ubuntu-14.04-amd64-1/  Systems Administrator's Guide Page 121

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

g-pan committed:
HPCC-17846 DOCS:Additional Info for Multiple Thors

Fix HPCC-17846 DOCS:Additional Info for Multiple Thors
Adding additional information to the section about running multiple Thors
on the same cluster in the System Administrators book.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 99ea11ca4ac3a661911223b75c08e9416a0dea62
Build: success
Install hpccsystems-platform-community_6.4.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

JamesDeFabia commented on docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:
```diff
@@ -1662,8 +1662,15 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
+        the Thors. This needs to be configured per Thor cluster definition.
+        </para>
+
+        <para>Multiple Thor's on the same cluster require them to share the
```
No apostrophe on plural form of Thor.

JamesDeFabia commented on docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:
```diff
@@ -1662,8 +1662,15 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
+        cluster, which can share the same machine set. There are slave and
+        master port settings that need to be set to avoid clashing. There are
+        also memory sharing/splitting considerations and settings that need to
+        be made. </para>
```
You say there are settings to be set to avoid clashing but don't say which ones. You also say there are memory considerations but don't say what those considerations are.

JamesDeFabia reviewed (changes requested):
One grammar typo plus some details that need further explanation.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: ff9e7a7287778116d317258cf4f1cd9461dfef6b
Build: success
Install hpccsystems-platform-community_6.4.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 8305e34aaf9cb20b185951ad439766bb968cc7be
Build: success
Install hpccsystems-platform-community_6.4.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 29e117ef80bd4deb59e8723876468a782cd0655a
Build: success
Install hpccsystems-platform-community_6.4.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 7c63bc10a80b1431cf4ff35e3a24838ba49bc49e
Build: success
Install hpccsystems-platform-community_6.4.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

JamesDeFabia reviewed (commented):
Looks good from my POV, but @jakesmith needs to have the last word

richardkchapman commented:
@jakesmith

jakesmith commented on docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:
```diff
@@ -1662,8 +1662,93 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
+                  role="bold">masterMemorySize</emphasis></entry>
+
+                  <entry>The maximum memory a Thor master can use. If left
+                  blank it will use the global value.</entry>
```
perhaps should say : ' If left blank it will use the globalMemorySize value. '

jakesmith commented on docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:
```diff
@@ -1662,8 +1662,93 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
+
+                  <entry>This value determines where the sprayed data goes and
+                  it must be the same for the same set of nodes (IP
+                  addresses).</entry>
```
I think I'd reword this as:

The nodeGroup is associated with files published by this Thor instance, normally it is left blank and defaults to the same as the 'name' attribute. In environments with multiple Thor's sharing the same group of nodes, the 'name' of each Thor must be different, but the nodeGroup value of all the Thor's sharing the same physical nodes should be set to the same name. NB: It is important to make the nodeGroup value equal to one of the Thor instance name values.

jakesmith commented:
@g-pan - a couple of comments

g-pan committed:
HPCC-17846 DOCS:Additional Info for Multiple Thors

Fix HPCC-17846 DOCS:Additional Info for Multiple Thors
Adding additional information to the section about running multiple Thors
on the same cluster in the System Administrators book.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

g-pan commented:
Updated to include @jakesmith comments. Thank you.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: eb8677368a98ca0cfa14ca92ca0a742f7fd8120a
Build: success
Install hpccsystems-platform-community_6.4.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
@g-pan - looks fine.

@richardkchapman - ready to merge.
